### DISCUSSION TITLE
support web

### DISCUSSION THREAD

**Discussion by direstrepo24** [buijs-dev/klutter#10] [General]
How would the plugin support for the web, so as not to rewrite all the logic, taking advantage of kmp kotlin?

**Comment by buijs-dev** [comment]
That wouldn't be too hard to do, but of course depends on your use case. 

The kmp module would have to need KotlinJS as a target and then Flutter web could use it as a npm package. Maybe it's possible to just output the raw JS code and let Flutter call it locally. 

I'm not sure if this works the same for web as it does for iOS/Android when building a plugin. Would have to look into it. Generating the boilerplate would be a lot easier for web than mobile so that won't be too hard.

**Reply by direstrepo24** [reply]
a ok, thanks. I think it would be something similar to what is indicated in this article, https://tbrgroup.software/writing-a-flutter-web-plugin-with-javascript/, but calling kotlinJS functions. 

I thought that in the end after getting the data from ios or android, there was no need to do anything special to send it to the web. It should be something like a generic viewmodel.

**Reply by buijs-dev** [reply]
That's definitely possible. The generated dart code should have a separate path of execution for JS methods for web but that could be added as a facade or something similar. So then you could just call the flutter plugin code the same way you would for mobile. 

Is that what you are looking for? 

I think it's interesting so I'll add it to the TODO list.

**Comment by direstrepo24** [comment]
yes. cool. Thanks!!
